packages/router/src/utils/config_matching.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Route} from '../config';
+import {defaultUrlMatcher, PRIMARY_OUTLET} from '../shared';
+import {UrlSegment, UrlSegmentGroup} from '../url_tree';
+
+import {forEach} from './collection';
+import {getOutlet} from './config';
+
+export interface MatchResult {
+  matched: boolean;
+  consumedSegments: UrlSegment[];
+  lastChild: number;
+  parameters: {[k: string]: string};
+  positionalParamSegments: {[k: string]: UrlSegment};
+}
+
+const noMatch = {
```

Comment:
Would it make sense to assign `MatchResult` type here to make sure the object is always in sync with `MatchResult` interface?
```suggestion
const noMatch: MatchResult = {
```